src/MUSI6106Exec/MUSI6106Exec.cpp

```diff
@@ -87,15 +123,24 @@ int main(int argc, char* argv[])
 
     time = clock();
 
+    //////////////////////////////////////////////////////////////////////////////
+    // initialize Comb Filter
+    CCombFilterIf::create(phCombFilter);
+    phCombFilter->init(eFilterType, fMaxDelay, stFileSpec.fSampleRateInHz, stFileSpec.iNumChannels);
+    phCombFilter->setParam(CCombFilterIf::FilterParam_t::kParamDelay, fDelayParam);
+    phCombFilter->setParam(CCombFilterIf::FilterParam_t::kParamGain, fGainParam);
+
     //////////////////////////////////////////////////////////////////////////////
     // get audio data and write it to the output text file (one column per channel)
+    // apply comb filter
     while (!phAudioFile->isEof())
     {
         // set block length variable
         long long iNumFrames = kBlockSize;
 
         // read data (iNumOfFrames might be updated!)
         phAudioFile->readData(ppfAudioData, iNumFrames);
+        phCombFilter->process(ppfAudioData, ppfAudioData, iNumFrames);
 
```

Comment:
The same issue as John has mentioned that you forgot to add something like `phOutputAudioFile->writeData(ppfOutputAudioData, iNumFrames)`. This is also part of the assignment requirement.